dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -103,10 +105,10 @@ class TeacherApplication < ApplicationBase
     # 1. Application has a specific school? always overwrite the user's school info
     # 2. User doesn't have a specific school? overwrite with the custom school info.
     def update_user_school_info!
-      if school_id || user.school_info.try(&:school).nil?
-        school_info = get_duplicate_school_info(school_info_attr) || SchoolInfo.create!(school_info_attr)
-        user.update_school_info(school_info)
```

Comment:
On an incomplete application, `school_info` may be nil, in which case we don't want to update anything on the user table. Added tests to document the new functionality and ensure old functionality doesn't change